doc/embed_compiled_native.md

```diff
@@ -106,3 +110,127 @@ gcc -O3 -o nosandbox_with_staticlib ../src/nosandbox_main.c -L. -lstaticnosandbo
 # Example usage of executable, in the nosandbox_main.c it calls the add function in the wasm object file
 ./nosandbox_with_staticlib
 ```
+
+## General workflow of embedding wasm object file generated library
+
+In the previous two sections, we have seen how to use the final product of the wasm app compiled to native in sandbox mode and no-sandbox mode. Here we will explore more details of the general workflow of embedding the wasm final product library in the host native binary.
+
+For the **no-sandbox mode** library, use the library as you would use a normal library, for example, in your wasm application you will make a library as the final product, and you can have such function:
+
+```C
+// wasm app C source
+int
+add(int a, int b)
+{
+    return a + b;
+}
+```
+
+Then in your host native binary, you can call this function directly.
+
+```C
+#include <stdio.h>
+
+extern int
+add(int a, int b);
+
+int
+main()
+{
+    printf("Nosandbox main: add 3 + 4 = %d\n", add(3, 4));
+}
+```
+
+So in this section, we will focus on the sandbox mode library.
+
+Here is the more detailed process of how to use the export APIs defined in `w2n_export.h` to interact with the wasm app(in the form of a library) in the host native binary:
+
+### Embed the wasm app library in the host native binary
+
+First, the wasm instance needs to be created using `wasm_instance_create()`, and check whether it creates successfully using `wasm_instance_is_created()`. If it fails or at the end of the host native binary, we need to destroy the wasm instance.
+
+```C
+    wasm_instance_create();
+    if (!wasm_instance_is_created()) {
+        os_printf("Create wasm instance failed.\n");
+        goto fail;
+    }
+
+    ...
+
+fail:
+    wasm_instance_destroy();
+    return ret;
+```
+
+Then you can use `wasm_get_export_apis()` and `wasm_get_export_api_num()` to get and call the wasm function by name, following similar procedures as below(a full example process can be found in `execute_func` of [wasm_application.c](wasm2native-vmlib/application/wasm_application.c)):
+
+```C
+    WASMExportApi *export_apis = wasm_get_export_apis();
+    WASMExportApi *export_func = NULL;
+    void (*invoke_native)(const void *, uint32 *, uint32 *);
+    uint32 export_api_num = wasm_get_export_api_num();
+
+    /* Lookup exported function */
+    for (i = 0; i < export_api_num; i++) {
+        if (!strcmp(export_apis[i].func_name, name)) {
+            export_func = &export_apis[i];
+            break;
+        }
+    }
+
+    /* from the export_func->signature, parse and prepare the arguments array(argv1) */
+    ...
+
+    /* lookup the quick aot entry, basically it will reinterpret the function ptr type that will match export_func->signature */
```

Comment:
lookup_quick_aot_entry is provided by wasm_application.c and is static. Developer must link the libcvmlib.a, but we suppose that he doesn't have to link it.

Suggest to just convert the func_ptr to an function according the prototype, and call it directly, like:
```C
    typedef int32 (*AOTFunc)(int32);
    AOTFunc aot_func = (AOTFunc)export_func->func_ptr;
    int32 result = aot_func(..);
```